pinot-common/src/main/java/org/apache/pinot/common/assignment/InstancePartitions.java

```diff
@@ -127,6 +127,13 @@ public ZNRecord toZNRecord() {
     return znRecord;
   }
 
+  /**
+   * Returns a new instance of InstancePartitions with the given name
+   */
+  public InstancePartitions withName(String newName) {
```

Comment:
(minor) I think we can just add `setInstancePartitionsName(String instancePartitionsName)` so that we can reuse the same object

Reply:
I added this out of an abundance of caution. Was concerned that it might be easy to run into buggy scenarios in the future where we store an instance partitions with name `foo` at node `/INSTANCE_PARTITIONS/bar`.

Reply:
Sounds good since it is not in performance critical path